src/Bridge/Symfony/Bundle/DataCollector/RequestDataCollector.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Bundle\DataCollector;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceNameCollectionFactoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+
+class RequestDataCollector extends DataCollector
+{
+    /**
+     * collectionFactory.
+     *
+     * @var ResourceNameCollectionFactoryInterface
+     */
+    private $collectionFactory;
+
+    /**
+     * metadataFactory.
+     *
+     * @var ResourceMetadataFactoryInterface
+     */
+    private $metadataFactory;
+
+    /**
+     * __construct.
+     *
+     * @param ResourceNameCollectionFactoryInterface $collectionFactory
+     * @param ResourceMetadataFactoryInterface       $metadataFactory
+     */
+    public function __construct(ResourceNameCollectionFactoryInterface $collectionFactory, ResourceMetadataFactoryInterface $metadataFactory)
+    {
+        $this->collectionFactory = $collectionFactory;
+        $this->metadataFactory = $metadataFactory;
+    }
+
+    public function collect(Request $request, Response $response, \Exception $exception = null)
+    {
+        $resourceClass = $request->attributes->get('_api_resource_class');
+        $resourceMetadata = $resourceClass ? $this->metadataFactory->create($resourceClass) : null;
+
+        $this->data = [
+            'resource_class' => $resourceClass,
+            'resource_metadata' => $resourceMetadata,
+            'method' => $request->getMethod(),
+            'acceptable_content_types' => $request->getAcceptableContentTypes(),
+        ];
+    }
+
+    public function getMethod()
```

Comment:
`: string`, same for all methods

Reply:
Added the best I can but as can when value maybe null I can't until php7.0 support is dropped